### PR TITLE
Release 2023-10 snapshot and update IDs to reflect updated McEliece in liboqs v0.9.0

### PR DESCRIPTION
**Pull request by praveksharma**
This updates the IDs to reflect the changes to McEliece in liboqs v0.9.0.

This updates release documentation for OQS-OpenSSH snapshot 2023-10.

### CODE REVIEW

**Comment by praveksharma** [comment]
I created a new branch since I'd misunderstood the branch naming scheme which led to delay in making this pull request.

**Review by baentsch — CHANGES_REQUESTED** [review]
Thanks for these ID updates, @praveksharma . What's missing now is running the script `generate.py` that reads the changes in the YML and puts them into the code. If you consider then doing a full release please also check out https://github.com/open-quantum-safe/openssh/wiki/Release-process (a bit brief, admitted, but arguably representative of how little attention this subproject is receiving).

**Comment by praveksharma** [comment]
Thank you for the feedback @baentsch. I've run `generate.py` and updated the relevant documentation for a full release. No changes to `version.h` have been made since the OpenSSH version hasn't changed.

**Review by baentsch — APPROVED** [review]
LGTM. Thanks for the update!

**Comment by praveksharma** [comment]
@baentsch, I was unsure how to interpret your comment in [open-quantum-safe/oqs-demos#237](https://github.com/open-quantum-safe/oqs-demos/pull/237):

> When [open-quantum-safe/openssh#148](https://github.com/open-quantum-safe/openssh/pull/148) gets merged (not by me but someone knowledgeable in OpenSSH)

Are you saying I should request someone other than you review this PR before I go ahead and merge it?

**Comment by baentsch** [comment]
> Are you saying I should request someone other than you review this PR before I go ahead and merge it?

I don't know enough how confident you feel (about judging whether this code is sufficiently well tested). Normally, the person raising the PR should also merge (if confident about the change). All I want to say is that I am not familiar enough with `openssh` to confidently say that everything's perfectly tested: I only looked at the diff (appear logical) and the CI results: The latter are "shallow" (not many platforms, not many config options tested) -- but the whole project is not "thoroughly maintained", soooo....

Long story short, go ahead and merge. I did the same when Goutam left -- besides, probably not too many people will scream if something's wrong :-)
